src/System.Private.CoreLib/shared/System/TimeZoneInfo.Unix.cs

```diff
@@ -406,85 +408,112 @@ private static string FindTimeZoneIdUsingReadLink(string tzFilePath)
             return id;
         }
 
+        private static string GetDirectoryEntryFullPath(ref Interop.Sys.DirectoryEntry dirent, string currentPath)
+        {
+            Span<char> nameBuffer = stackalloc char[Interop.Sys.DirectoryEntry.NameBufferSize];
+            ReadOnlySpan<char> direntName = dirent.GetName(nameBuffer);
+
+            if ((direntName.Length == 1 && direntName[0] == '.') ||
+                (direntName.Length == 2 && direntName[0] == '.' && direntName[1] == '.'))
+                return null;
+
+            return Path.Join(currentPath.AsSpan(), direntName);
+        }
+
         /// <summary>
         /// Enumerate files
         /// </summary>
-        private static IEnumerable<string> EnumerateFilesRecursively(string path)
+        private static unsafe string FindFile(string path, Predicate<string> condition)
```

Comment:
The string that this returns is never used. I would keep `EnumerateFilesRecursively` name and make it return `void`.